core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java

```diff
@@ -148,6 +148,39 @@ public void testScriptsWithoutExtensions() throws IOException {
         }
     }
 
+    public void testHiddenFileSkipped() throws IOException {
+        ContextAndHeaderHolder contextAndHeaders = new ContextAndHeaderHolder();
+        buildScriptService(Settings.EMPTY);
+
+        logger.info("--> setup one hidden test file");
+        Path testFileHidden = scriptsFilePath.resolve(".hidden_file");
+        Path testRegularFile = scriptsFilePath.resolve("test_file.tst");
+        Streams.copy("test_hidden_file".getBytes("UTF-8"), Files.newOutputStream(testFileHidden));
+        Streams.copy("test_file".getBytes("UTF-8"), Files.newOutputStream(testRegularFile));
+        resourceWatcherService.notifyNow();
+
+        try {
+            logger.info("--> verify if hidden_file was skipped");
+            scriptService.compile(new Script("hidden_file", ScriptType.FILE, "test", null),
```

Comment:
Hi @jasontedor, 
I was wondering if we could just change the method:
`private Tuple<String, String> scriptNameExt(Path file)` from `ScriptChangesListener` to ignore any kind of file that does not have a name. I mean, hidden files. 

It starts with `.` but the method allows you to retrieve an empty script name for this case. If we just use the `if (extIndex > 0)` we would ignore this kind of file. In addition I would move the method to the parent class and drop the access modifier to create the unit test. We would not need to check if the file is a hidden file or not.
Please, let me know what do you think. 
Thanks